app/bundles/LeadBundle/Helper/IdentifyCompanyHelper.php

```diff
@@ -129,14 +129,18 @@ public static function findCompany(array $parameters, CompanyModel $companyModel
      */
     protected static function domainExists($email)
     {
-        list($user, $domain) = explode('@', $email);
-        $arr                 = dns_get_record($domain, DNS_MX);
+        if (!strstr($email, '@')) { //not a valid email adress
+            return false;
+        } else {
+            list($user, $domain) = explode('@', $email);
+            $arr                 = dns_get_record($domain, DNS_MX);
 
-        if ($arr && $arr[0]['host'] === $domain) {
-            return $domain;
+            if ($arr && $arr[0]['host'] === $domain) {
```

Comment:
It's a little bit cleaner here to check if $arr is empty and if so, return false. All the if/elses make it a bit tricky to read what's going on. 

```
if (empty($arr)) {
    return false;
}

if ($arr[0]['host'] === $domain) {
    return $domain;
}

return false;
```

Reply:
I've made all the changes you required. Thanks for your advices.